README.md

```diff
@@ -1,17 +1,80 @@
-## My Project
+# A web site and REST API with Cognito authentication (user pools) using the Facebook identity provider
 
-TODO: Fill this README out!
+The example will show you how to create the following:
 
-Be sure to:
+- A single-page app hosted by S3 and CloudFront
+- A REST API that uses Cognito for authentication
+- Integration of Facebook as an identity provider
 
-* Change the title in this README
-* Edit your repository description on GitHub
+It also demonstrates a somewhat opinionated way to organize your lambda functions and test them.
 
-## Security
+## Prerequisites
 
-See [CONTRIBUTING](CONTRIBUTING.md#security-issue-notifications) for more information.
+There is a bit of setup required before you can deploy this stack.
 
-## License
+- A domain that you control with Route53 in your development account
+    - https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/getting-started.html
+    - Be careful about Cognito reserved domains.
+        - You can't have 'aws' or 'cognito' anywhere in the domain name
+- A certicate that covers the web site and the API, created with ACM
+    - For example, you own "example.com"
+    - You have a certificate for "myapp.example.com" and "myapi.example.com"
+    - https://docs.aws.amazon.com/acm/latest/userguide/gs.html
+- A Facebook Developer account
+    - Create an app and put the app id into your .env
+    - Store the app secret in AWS Secrets Manager as facebook_app_secret
+    - https://developers.facebook.com/docs/facebook-login/
+    - https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-social-idp.html
+    - (Don't follow the separate guide for integrating Facebook with Identity Pools)
 
-This library is licensed under the MIT-0 License. See the LICENSE file.
+## Environment Variables
 
+Create file called config/env-local.json using the following json as a template. Replace those values with your own.
+(Some of the values, starting with cognitoPoolId, are only used for integration testing and can be left blank until after your first deployment)
+
+```json
+{ 
+    "env": {
+        "account": "012345678901",
+        "region": "us-east-1"
+    }, 
```

Comment:
Still curious why do it this way instead of asking the user to configure it directly in code at the entry point of `App` in `cognito-idp.ts`?

Reply:
Because you may have a large team of developers working on an app, and you wouldn't want to put all of their dev environment variables into source.